handlers/deploy.go

```diff
@@ -118,21 +127,41 @@ func MakeDeployHandler(functionNamespace string, clientset *kubernetes.Clientset
 
 func makeDeploymentSpec(request requests.CreateFunctionRequest, existingSecrets map[string]*apiv1.Secret, config *DeployHandlerConfig) (*v1beta1.Deployment, error) {
 	envVars := buildEnvVars(&request)
-	path := filepath.Join(os.TempDir(), ".lock")
-	probe := &apiv1.Probe{
-		Handler: apiv1.Handler{
+	var handler apiv1.Handler
+
+	if config.HTTPProbe {
```

Comment:
This looks good for now but in the future we should have a way to switch this flag from the function definition so that is backwards compatible with functions that are built with the old watchdog.

Reply:
For instance, we could use the new annotations field being worked on by @ewilde